dashboard/config/initializers/script_preload.rb

```diff
@@ -7,11 +7,11 @@
     Script.should_cache? &&
     !Rails.application.config.skip_script_preload
   # Populate the shared in-memory cache from the database.
-  Script.unit_cache_to_cache
+  Script.unit_cache_to_cache unless Rails.cache.is_a?(ActiveSupport::Cache::MemoryStore)
   Script.script_cache
   Script.script_level_cache
   Script.level_cache
   Script.unit_family_cache
-  UnitGroup.course_cache_to_cache
+  UnitGroup.course_cache_to_cache unless Rails.cache.is_a?(ActiveSupport::Cache::MemoryStore)
```

Comment:
@wjordan what purpose do the new checks in this file serve?

Reply:
It relates to the changes in [`UnitGroup.course_cache_to_cache`](https://github.com/code-dot-org/code-dot-org/pull/41909/files#diff-e2172d22099b0f25366646fc780a03cad5198af1e0f0e74b87c58499558e4ce4R614-R616) and [`Script.unit_cache_to_cache`](https://github.com/code-dot-org/code-dot-org/pull/41909/files#diff-22769ab80ff34eaf4aadc1dc87e1b9bdad8b445cec233c9a5a1d2bcc5a7a7978L394-R401).

The observation was that this initializer was loading the curriculum hierarchy into a Ruby-memory object, writing it to a Cache object, then reading it back from Cached object to a Ruby-memory object. In the case when the Cache is itself a Ruby-memory object, this back and forth was entirely redundant and could be avoided.

I didn't profile the startup-performance difference on this tweak, and it's separate from the rest of the changes in this PR so could be easily left out if seems risky or causes issues. Maybe there's a better way to express the general idea with some further refactoring as well.